test/model/src/driver/kv_workload_runner_wt.cpp

```diff
@@ -0,0 +1,377 @@
+    if (ret != 0)
+        throw wiredtiger_exception("Failed to open a cursor", ret);
+
+    _cursors[id] = cursor;
+    return cursor;
+}
+
+/*
+ * kv_workload_runner_wt::~kv_workload_runner_wt --
+ *     Clean up the workload context.
+ */
+kv_workload_runner_wt::~kv_workload_runner_wt()
+{
+    if (_connection != nullptr) {
+        try {
+            wiredtiger_close();
+        } catch (std::runtime_error &e) {
+            /*
+             * We cannot throw an exception out of a destructor, so just print a warning and
+             * continue.
+             */
+            std::cerr << "Error while cleaning up the workload context: " << e.what() << std::endl;
+        }
+    }
+}
+
+/*
+ * kv_workload_runner_wt::do_operation --
+ *     Execute the given workload operation in WiredTiger.
+ */
+int
+kv_workload_runner_wt::do_operation(const operation::begin_transaction &op)
+{
+    std::shared_lock lock(_connection_lock);
+    session_context_ptr session = allocate_txn_session(op.txn_id);
+    return session->session()->begin_transaction(session->session(), nullptr);
+}
+
+/*
+ * kv_workload_runner_wt::do_operation --
+ *     Execute the given workload operation in WiredTiger.
+ */
+int
+kv_workload_runner_wt::do_operation(const operation::checkpoint &op)
+{
+    std::shared_lock lock(_connection_lock);
+
+    WT_SESSION *session;
+    int ret = _connection->open_session(_connection, nullptr, nullptr, &session);
+    if (ret != 0)
+        throw wiredtiger_exception("Failed to open a session", ret);
+    wiredtiger_session_guard session_guard(session);
+
+    std::ostringstream config;
+    if (op.name != "")
+        config << "name=" << op.name;
+
+    std::string config_str = config.str();
+    return session->checkpoint(session, config_str.c_str());
+}
+
+/*
+ * kv_workload_runner_wt::do_operation --
+ *     Execute the given workload operation in WiredTiger.
+ */
+int
+kv_workload_runner_wt::do_operation(const operation::commit_transaction &op)
+{
+    std::shared_lock lock(_connection_lock);
+    session_context_ptr session = remove_txn_session(op.txn_id);
+
+    std::ostringstream config;
+    if (op.commit_timestamp != k_timestamp_none)
+        config << ",commit_timestamp=" << std::hex << op.commit_timestamp;
+    if (op.durable_timestamp != k_timestamp_none)
+        config << ",durable_timestamp=" << std::hex << op.durable_timestamp;
+
+    std::string config_str = config.str();
+    return session->session()->commit_transaction(session->session(), config_str.c_str());
+}
+
+/*
+ * kv_workload_runner_wt::do_operation --
+ *     Execute the given workload operation in WiredTiger.
+ */
+int
+kv_workload_runner_wt::do_operation(const operation::create_table &op)
+{
+    std::shared_lock lock(_connection_lock);
+
+    WT_SESSION *session;
+    int ret = _connection->open_session(_connection, nullptr, nullptr, &session);
+    if (ret != 0)
+        throw wiredtiger_exception("Failed to open a session", ret);
```

Comment:
Do we want to throw or simply return here? I can see we return `ret` at the end of the function.

Reply:
Yes, I guess that makes sense. I changed it here, and also above for the checkpoint operation.